packages/core-ethereum/crates/core-ethereum-rpc/src/rpc.rs

```diff
@@ -0,0 +1,381 @@
+use async_trait::async_trait;
+use core_crypto::keypairs::{ChainKeypair, Keypair};
+use core_crypto::types::Hash;
+use core_ethereum_types::{ContractAddresses, ContractInstances};
+use ethers::middleware::{MiddlewareBuilder, NonceManagerMiddleware, SignerMiddleware};
+use ethers::prelude::k256::ecdsa::SigningKey;
+use ethers::prelude::transaction::eip2718::TypedTransaction;
+use ethers::signers::{LocalWallet, Signer, Wallet};
+use ethers::types::BlockId;
+use ethers_providers::{JsonRpcClient, Middleware, Provider, RetryClient, RetryClientBuilder, RetryPolicy};
+use primitive_types::H160;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use std::time::Duration;
+use utils_types::primitives::{Address, Balance, BalanceType, U256};
+use validator::Validate;
+
+use crate::errors::Result;
+use crate::HoprRpcOperations;
+
+#[cfg(feature = "prometheus")]
+use utils_metrics::metrics::SimpleCounter;
+
+#[cfg(feature = "prometheus")]
+lazy_static::lazy_static! {
+     pub(crate) static ref METRIC_COUNT_RPC_CALLS: SimpleCounter = SimpleCounter::new(
+        "core_ethereum_counter_rpc_calls",
+        "Number of RPC calls"
+    )
+    .unwrap();
+}
+
+/// Configuration of the RPC related parameters.
+#[derive(Clone, Debug, Copy, PartialEq, Eq, Serialize, Deserialize, Validate)]
+pub struct RpcOperationsConfig {
+    /// Blockchain id
+    /// Default is 100.
+    pub chain_id: u64,
+    /// Addresses of all deployed contracts
+    /// Default contains empty (null) addresses.
+    pub contract_addrs: ContractAddresses,
```

Comment:
This is needed to create a filter at the indexer level... there should only be 1 place to have this config eventually... HoprChain. What are the contract addresses use for in the rpc? Withdraw?

Reply:
They are used to instantiate the Contracts which are used for contract calls later on.